integration/flink-native-listener/src/main/java/io/openlineage/flink/wrapper/TableLineageDatasetImplWrapper.java

```diff
@@ -0,0 +1,72 @@
+/*
+/* Copyright 2018-2024 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+
+package io.openlineage.flink.wrapper;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.Optional;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.reflect.MethodUtils;
+import org.apache.flink.streaming.api.lineage.LineageDataset;
+import org.apache.flink.table.catalog.CatalogBaseTable;
+import org.apache.flink.table.planner.lineage.TableLineageDatasetImpl;
+
+/**
+ * There is an issue that although {@link TableLineageDatasetImpl} is returned in a lineage graph,
```

Comment:
Agree on that. I was able to remove workaround and reflection with:
```
testImplementation("org.apache.flink:flink-table-test-utils:$flinkVersion") {
        exclude(group: "org.apache.flink", module: "flink-table-planner-loader")
}
testImplementation "org.apache.flink:flink-table-planner_2.12:$flinkVersion"
```

I think the problem still exists. Based on the [docs](https://nightlies.apache.org/flink/flink-docs-master/docs/dev/configuration/advanced/#table-planner-and-table-planner-loader), this will not work on default runtime. Docs says flink-table-planner-loader is loaded by default and contains the query planner hidden behind an isolated classpath (you won't be able to address any io.apache.flink.table.planner directly). Flink community is verifying this.